### PR TITLE
Add GitHub action to automatically update docs version

### PR DESCRIPTION
**Pull request by rnorth**

### CODE REVIEW

File: .github/workflows/update-docs-version.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        types: [published]
+
+jobs:
+    build:
+        runs-on: ubuntu-18.04
+        steps:
+            -   uses: actions/checkout@v1
+            -   uses: webfactory/ssh-agent@ea39f521c5e59e2e0a73686497d303bb2b499e03
+                                            # ^ v0.1.1
+                    with:
```
Comment (bsideup) [inline .github/workflows/update-docs-version.yml]: wrong indentation :D

File: .github/workflows/update-docs-version.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+on:
+    release:
+        types: [published]
+
+jobs:
+    build:
+        runs-on: ubuntu-18.04
+        steps:
+            -   uses: actions/checkout@v1
+            -   uses: webfactory/ssh-agent@ea39f521c5e59e2e0a73686497d303bb2b499e03
```
Comment (bsideup) [inline .github/workflows/update-docs-version.yml]: WDYT about using https://github.com/ad-m/github-push-action or similar action?

File: .github/workflows/update-docs-version.yml (hunk 1 above)
Comment (rnorth) [reply .github/workflows/update-docs-version.yml]: FFS YAML

File: .github/workflows/update-docs-version.yml (hunk 2 above)
Comment (rnorth) [reply .github/workflows/update-docs-version.yml]: This action, and others like it, all seem to use access tokens. AFAICT there is no way for us to create an access token that belongs to an organisation or repo; it's always a personal access token associated with an individual user.

I'd prefer not to use a personal access token for an automation, and deploy keys (SSH based) seem like the only way to create a credential that is effectively owned by the repository (I have discarded the private key after setting it as a secret for the repo).

Comment (bsideup) [reply .github/workflows/update-docs-version.yml]: the default access token that GHA assigns to every build automatically as a secret allows us to push to the same repo AFAIK

Comment (rnorth) [reply .github/workflows/update-docs-version.yml]: I thought that disallowed pushes... will check.

Comment (rnorth) [reply .github/workflows/update-docs-version.yml]: Actually, looks like I might have been wrong. Will give it a try.

File: .github/workflows/update-docs-version.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        runs-on: ubuntu-18.04
+        steps:
+            -   uses: actions/checkout@v1
+            -   name: Update latest_version property in mkdocs.yml
+                run: |
+                    sed -i "s/latest_version: .*/latest_version: ${GITHUB_REF##*/}/g" mkdocs.yml
+            -   name: Commit changes
+                env:
+                    COMMIT_MSG: |
+                        Update documentation version to ${GITHUB_REF##*/}
```
Comment (bsideup) [inline .github/workflows/update-docs-version.yml]: note that `${GITHUB_REF##*/}` is bash's syntax, not GHA, and the commit message will not have a version in it, but the placeholder

File: .github/workflows/update-docs-version.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                        skip-checks: true
+                run: |
+                    git config user.email "nobody@testcontainers.org"
+                    git config user.name "GitHub Action - Update docs version"
+                    git checkout master
```
Comment (bsideup) [inline .github/workflows/update-docs-version.yml]: already checked out by `actions/checkout@v1`

File: .github/workflows/update-docs-version.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                    git add .
```
Comment (bsideup) [inline .github/workflows/update-docs-version.yml]: Just to be safe:
```suggestion
                    git add mkdocs.yml
```

File: .github/workflows/update-docs-version.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+                    # Only commit and push if we have changes
+                    git diff && git diff --staged --quiet || (git commit -m "${COMMIT_MSG}")
```
Comment (bsideup) [inline .github/workflows/update-docs-version.yml]: WDYT about:
```suggestion
                    git commit -m "${COMMIT_MSG}" || echo "not changed"
```
?

File: .github/workflows/update-docs-version.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        types: [published]
+
+jobs:
+    build:
+        runs-on: ubuntu-18.04
+        steps:
+            -   uses: actions/checkout@v1
+            -   name: Update latest_version property in mkdocs.yml
+                run: |
+                    sed -i "s/latest_version: .*/latest_version: ${GITHUB_REF##*/}/g" mkdocs.yml
```
Comment (bsideup) [inline .github/workflows/update-docs-version.yml]: 1) this should go after `git checkout master`
2) idea: add `git diff` after `sed` for debugging

File: .github/workflows/update-docs-version.yml [hunk 8]
```diff
@@ -9,21 +9,18 @@ jobs:
...
-                        skip-checks: true
                 run: |
                     git config user.email "nobody@testcontainers.org"
                     git config user.name "GitHub Action - Update docs version"
-                    git checkout master
-                    git add .
-                    # Only commit and push if we have changes
-                    git diff && git diff --staged --quiet || (git commit -m "${COMMIT_MSG}")
+                    git add mkdocs.yml
+                    git commit -m "Update documentation version to ${GITHUB_REF##*/}" -m "skip-checks: true" || echo "not changed"
```
Comment (bsideup) [inline .github/workflows/update-docs-version.yml]: double (`-m`) message?

Comment (rnorth) [reply .github/workflows/update-docs-version.yml]: Interestingly, it’s both valid and git CLI’s way to create a multi-line commit message. It was a learning for me, but it does what we need, without the faff of escaping a `\n` in the string 😄

**Pull request merged**